src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -223,7 +227,12 @@ public bool TrySliceTo(byte b1, out ReadableBuffer slice, out ReadCursor cursor)
         public ReadableBuffer Slice(int start, int length)
         {
             var begin = _start.Seek(start);
-            return Slice(begin, begin.Seek(length));
+            var end = begin.Seek(length);
```

Comment:
There is a check in seek to make sure we are not Seeking past the end of segments, but we need also to check that we are not past the `End`. I was thinking about passing End to Seek and handling everything at one place.